src/script/miniscript.h

```diff
@@ -0,0 +1,1561 @@
+ *   - When satisfied, pushes a nonzero value (like B) on top of the stack, or one below.
+ *   - When dissatisfied, pushes 0 op top of the stack or one below.
+ *   - Is always "OP_SWAP [B]" or "OP_TOALTSTACK [B] OP_FROMALTSTACK".
+ *   - For example sc:pk_k(key) = OP_SWAP <key> OP_CHECKSIG
+ *
+ * There a type properties that help reasoning about correctness:
+ * - "z" Zero-arg:
+ *   - Is known to always consume exactly 0 stack elements.
+ *   - For example after(n) = <n> OP_CHECKLOCKTIMEVERIFY
+ * - "o" One-arg:
+ *   - Is known to always consume exactly 1 stack element.
+ *   - Conflicts with property 'z'
+ *   - For example sha256(hash) = OP_SIZE 32 OP_EQUALVERIFY OP_SHA256 <hash> OP_EQUAL
+ * - "n" Nonzero:
+ *   - For every way this expression can be satisfied, a satisfaction exists that never needs
+ *     a zero top stack element.
+ *   - Conflicts with property 'z' and with type 'W'.
+ * - "d" Dissatisfiable:
+ *   - There is an easy way to construct a dissatisfaction for this expression.
+ *   - Conflicts with type 'V'.
+ * - "u" Unit:
+ *   - In case of satisfaction, an exact 1 is put on the stack (rather than just nonzero).
+ *   - Conflicts with type 'V'.
+ *
+ * Additional type properties help reasoning about nonmalleability:
+ * - "e" Expression:
+ *   - This implies property 'd', but the dissatisfaction is nonmalleable.
+ *   - This generally requires 'e' for all subexpressions which are invoked for that
+ *     dissatifsaction, and property 'f' for the unexecuted subexpressions in that case.
+ *   - Conflicts with type 'V'.
+ * - "f" Forced:
+ *   - Dissatisfactions (if any) for this expression always involve at least one signature.
+ *   - Is always true for type 'V'.
+ * - "s" Safe:
+ *   - Satisfactions for this expression always involve at least one signature.
+ * - "m" Nonmalleable:
+ *   - For every way this expression can be satisfied (which may be none),
+ *     a nonmalleable satisfaction exists.
+ *   - This generally requires 'm' for all subexpressions, and 'e' for all subexpressions
+ *     which are dissatisfied when satisfying the parent.
+ *
+ * One type property is an implementation detail:
+ * - "x" Expensive verify:
+ *   - Expressions with this property have a script whose last opcode is not EQUAL, CHECKSIG, or CHECKMULTISIG.
+ *   - Not having this property means that it can be converted to a V at no cost (by switching to the
+ *     -VERIFY version of the last opcode).
+ *
+ * Five more type properties for representing timelock information. Spend paths
+ * in miniscripts containing conflicting timelocks and heightlocks cannot be spent together.
+ * This helps users detect if miniscript does not match the semantic behaviour the
+ * user expects.
+ * - "g" Whether the branch contains a relative time timelock
+ * - "h" Whether the branch contains a relative height timelock
+ * - "i" Whether the branch contains a absolute time timelock
+ * - "j" Whether the branch contains a absolute time heightlock
```

Comment:
If you touch this perhaps worth changing to "absolute height timelock"

Reply:
mini-nit: It should be "a**n** absolute [...]" in both lines